src/engine.js

```diff
@@ -143,9 +149,24 @@ Engine.prototype.off = function (event, callback, context) {
  *
  */
 Engine.prototype.reload = function (options) {
+  var self = this;
```

Comment:
I don't understand well this function.

1.- Why don't you use the `debounce` function from `underscore`?

2.- I don't grasp the use of Promise here since `_reload` already returns one.  Let me check it in more detail.